src/App.tsx

```diff
@@ -2,71 +1,94 @@
+import React, { useState, useMemo, useCallback, useRef, useEffect } from 'react';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownMenu } from './components/DropdownMenu';
+
+
+function debounce(callback: Function, delay: number = 300) {
+  let timerId: number = 0;
+
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+    timerId = window.setTimeout(() => {
+      callback(...args)
+    }, delay)
+  };
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [name, setName] = useState('');
+  const [appliedPerson, setAppliedPerson] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const applyPerson = useMemo(() => debounce(setAppliedPerson, 1000), []);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter((person) =>
+      person.name.toLowerCase().includes(appliedPerson.toLowerCase()),
```

Comment:
The filter should not trigger on spaces only. Consider trimming the input value and checking if it's not empty before filtering.